integration-tests/validator-tool_test.go

```diff
@@ -0,0 +1,102 @@
+package integrationtests
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	binaryName = "configuration-validator"
+)
+
+func TestMain(m *testing.M) {
+	// TODO I could not find a more robust way of reaching the root of the repo
```

Comment:
may be adding an env variable will help find package path?

Reply:
it's not so much about the package path, but we need to make sure to sure `make` from the root of the repo. At some point during this PR I relied on an env var called `GOREPO` (which relied on `GOPATH`), but since `GOPATH` is [away](https://blog.golang.org/modules2019) I don't think we should use that solution. Do you think it's worth having another env var in our makefile instead? On one hand I'm not fond of the current solution for it can easily break if we decide to move the integration tests to a sub-folder, OTOH, not having another dependency on the environment is better. What are the odds of us moving the config-validator integration tests file?

Reply:
i might be missing the point, why can't we run the `make compile-validator-tool` in the github workflow?

Reply:
@jyotimahapatra , that's actually an amazing idea!

Reply:
on a second thought, we're just shifting the problem from one place to another, i.e., we still need to be at the root of the repo to run the tool in the integration tests.

Reply:
I mean, we could still run `make compile-validator-tool` in the github workflow, but we still need a way to find the binary location in the tests.

Reply:
thinking about this some more. may be use `make compile-validator-tool` in gh workflow and then place the binary in `/usr/local/bin` . In the test reference the path directly instead of `chDir`.

Reply:
I tried this in https://github.com/envoyproxy/xds-relay/pull/32/checks?check_run_id=537737007 and realized that the path `/usr/local/bin` is not available in our github actions workflow. In fact, if we wanted to set it up we'd need to setup all the paths (following something like https://stackoverflow.com/questions/59236611/github-actions-run-golint-on-wihdows/59242962#59242962). But given that we're not investing in relying on GOPATH (since it was deprecated [as of go 1.13](https://blog.golang.org/modules2019)), I don't think we should rely no it.

What do you think?

Reply:
That was a bummer. Agree that we shouldn't use `GOPATH`. 
I think we have exhausted a lot of approaches. The last one i can think of is still use make from gh actions and place the file in a relative location in the repo based on a env var. pretty gnarly?

Reply:
at this point I'm not sure we're doing anything better than relying on the current solution (i.e. relying on the relative path).

Reply:
true. If it makes sense we can put some of this info in the comment. But not a blocker.